### DISCUSSION TITLE
Arduino_Core_STM32  not compatible with Arduino Modbus library

### DISCUSSION THREAD

**Discussion by smn018** [stm32duino/Arduino_Core_STM32#2358] [Libraries]
I used  Arduino_Core_STM32  V1.9.0. Recently I updated the version of “ Arduino_Core_STM32 ” from V1.9.0 to V2.7.1 . Ardunio ide tip： ArduinoModbus and ArduinoRS485 not compatible with STM32. I compile the old project and then use jlink to flash .bin file to stm32F103C8T6.  The RS485 communication couldn't work any more. How can I solve this problem?

**Comment by fpistm** [comment]
Hi @smn018 
Don't know why it work before and not now. Maybe timing issue or other.
Looking at the libraries issues, it seems some concerned STM32.
Ex:
https://github.com/arduino-libraries/ArduinoRS485/issues/9
https://github.com/arduino-libraries/ArduinoModbus/issues/47 (even if it concerns Roger core )

Probably link to the way Serial is used by the library, when USB is enabled the Serial is mapped to SerialUSB.
But can't do nothing for this.
